contracts/SportMarkets/Parlay/ParlayVerifier.sol

```diff
@@ -552,115 +542,37 @@ contract ParlayVerifier {
-        returns (
-            uint totalQuote,
-            uint totalBuyAmount,
-            uint skewImpact,
-            uint[] memory finalQuotes,
-            uint[] memory buyAmountPerMarket
-        )
-    {
-        uint[] memory buyQuoteAmountPerMarket = new uint[](params.sportMarkets.length);
-        buyAmountPerMarket = params.buyQuoteAmounts;
-        finalQuotes = new uint[](params.sportMarkets.length);
-        uint feesIncluded;
-        for (uint i = 0; i < params.sportMarkets.length; i++) {
-            totalBuyAmount += params.buyQuoteAmounts[i];
-            // buyQuote always calculated with added SportsAMM fees
-            buyQuoteAmountPerMarket[i] = (params.defaultONE *
-                params.sportsAmm.buyFromAmmQuote(
-                    params.sportMarkets[i],
-                    obtainSportsAMMPosition(params.positions[i]),
-                    params.buyQuoteAmounts[i]
-                ));
-            if (buyQuoteAmountPerMarket[i] == 0) {
-                totalQuote = 0;
-                totalBuyAmount = 0;
-            }
-        }
-        for (uint i = 0; i < params.sportMarkets.length; i++) {
-            finalQuotes[i] = ((buyQuoteAmountPerMarket[i] * ONE * ONE) / params.buyQuoteAmounts[i]) / ONE;
-            // where the SGP fee is actually applied
-            if (params.sgpFees[i] > 0) {
-                finalQuotes[i] = ((finalQuotes[i] * ONE * ONE) / params.sgpFees[i]) / ONE;
-                feesIncluded += params.sgpFees[i] > ONE ? params.sgpFees[i] - ONE : (ONE - params.sgpFees[i]);
-            }
-            totalQuote = (i == 0) ? finalQuotes[i] : (totalQuote * finalQuotes[i]) / ONE;
-        }
-        if (totalQuote > 0) {
-            if (totalQuote < IParlayMarketsAMM(params.sportsAmm.parlayAMM()).maxSupportedOdds()) {
-                totalQuote = IParlayMarketsAMM(params.sportsAmm.parlayAMM()).maxSupportedOdds();
-            }
-            uint expectedPayout = ((params.sUSDAfterFees * ONE * ONE) / totalQuote) / ONE;
-            skewImpact = expectedPayout > totalBuyAmount
-                ? (((ONE * expectedPayout) - (ONE * totalBuyAmount)) / (totalBuyAmount))
-                : (((ONE * totalBuyAmount) - (ONE * expectedPayout)) / (totalBuyAmount));
-            buyAmountPerMarket = _applySkewImpactBatch(buyAmountPerMarket, skewImpact, (expectedPayout > totalBuyAmount));
-            totalBuyAmount = _applySkewImpact(totalBuyAmount, skewImpact, (expectedPayout > totalBuyAmount));
-            _calculateRisk(params.sportMarkets, (totalBuyAmount - params.sUSDAfterFees), params.sportsAmm.parlayAMM());
-            if (feesIncluded > 0) {
-                if (skewImpact > feesIncluded) {
-                    skewImpact = skewImpact - feesIncluded;
-                } else {
-                    skewImpact = feesIncluded - skewImpact;
-                }
+            for (uint i = 0; i < numOfMarkets; i++) {
+                amountsToBuy[i] = (totalBuyAmount * finalQuotes[i]) / ONE;
```

Comment:
Not sure if we need this loop, since no amounts would be bought out of SportsAMM. 

@vpoklopic does the front need it on the subgraph or on the ticket panel prior to buy ?

Reply:
Position amounts on the parlay?

Reply:
> Position amounts on the parlay?

yes

Reply:
No, we show just quotes.